tracer/src/Datadog.Trace/Ci/Test.cs

```diff
@@ -0,0 +1,270 @@
+// <copyright file="Test.cs" company="Datadog">
+// Unless explicitly stated otherwise all files in this repository are licensed under the Apache 2 License.
+// This product includes software developed at Datadog (https://www.datadoghq.com/). Copyright 2017 Datadog, Inc.
+// </copyright>
+#nullable enable
+
+using System;
+using System.Collections.Generic;
+using System.Reflection;
+using System.Threading;
+using Datadog.Trace.Ci.Tagging;
+using Datadog.Trace.Ci.Tags;
+using Datadog.Trace.ExtensionMethods;
+using Datadog.Trace.Pdb;
+
+namespace Datadog.Trace.Ci;
+
+/// <summary>
+/// CI Visibility test
+/// </summary>
+public sealed class Test
+{
+    private static readonly AsyncLocal<Test?> CurrentTest = new();
+    private readonly Scope _scope;
+    private int _finished;
+
+    internal Test(TestSuite suite, string name, DateTimeOffset? startDate = null)
+    {
+        Suite = suite;
+        var module = suite.Module;
+
+        var tags = new TestSpanTags(Suite.Tags, name);
+        var scope = Tracer.Instance.StartActiveInternal(
+            string.IsNullOrEmpty(module.Framework) ? "test" : $"{module.Framework!.ToLowerInvariant()}.test",
+            tags: tags,
+            startTime: startDate);
+
+        scope.Span.Type = SpanTypes.Test;
+        scope.Span.ResourceName = $"{suite.Name}.{name}";
+        scope.Span.SetTraceSamplingPriority(SamplingPriority.AutoKeep);
+        scope.Span.SetTag(Trace.Tags.Origin, TestTags.CIAppTestOriginName);
+
+        _scope = scope;
+
+        if (CIVisibility.Settings.CodeCoverageEnabled == true)
+        {
+            Coverage.CoverageReporter.Handler.StartSession();
+        }
+
+        CurrentTest.Value = this;
+        CIVisibility.Log.Debug("######### New Test Created: {name} ({suite} | {module})", Name, Suite.Name, Suite.Module.Name);
+
+        if (startDate is null)
+        {
+            // If a test doesn't have a fixed start time we reset it before running the test code
+            scope.Span.ResetStartTime();
+        }
+    }
+
+    /// <summary>
+    /// Test status
+    /// </summary>
+    public enum Status
+    {
+        /// <summary>
+        /// Pass test status
+        /// </summary>
+        Pass,
+
+        /// <summary>
+        /// Fail test status
+        /// </summary>
+        Fail,
+
+        /// <summary>
+        /// Skip test status
+        /// </summary>
+        Skip
+    }
```

Comment:
From a public-API point of view, I don't think this should be a nested type?

Reply:
cool, done in [63aa306](https://github.com/DataDog/dd-trace-dotnet/pull/3276/commits/63aa306f68d4dd73248fdc5629b8b2fe9d01f4aa)